Code/Common/include/sitkCompositeTransform.h

```diff
@@ -0,0 +1,191 @@
+ * optimizable for registration.
+ *
+ * Inverse:
+ * The inverse transform is created by retrieving the inverse from each
+ * sub transform and adding them to a composite transform in reverse order.
+ * The m_TransformsToOptimizeFlags is copied in reverse for the inverse.
+ *
+ * \sa itk::CompositeTransform
+ */
+class SITKCommon_EXPORT CompositeTransform
+    : public Transform {
+public:
+  using Self = CompositeTransform;
+  using Superclass = Transform;
+
+  /** \brief Construct an empty CompositeTransform
+   *
+   *  The created CompositeTransform is initialized with zero transforms.
+   *  Additional transforms of dimensions can be added.
+   */
+  explicit CompositeTransform(unsigned int dimensions);
+
+  /** \brief Create CompositeTransform converted or holding the transform argument.
+   *
+   * If the Transform is internally a CompositeTransform, a shallow copy to
+   * the internal transform will be made. Otherwise a new CompositeTransform is
+   * constructed which holds the transform argument.
+   *
+   */
+  CompositeTransform( const Transform & );
+
+  /** \brief A lazy copy constructor
+   *
+   * The new SimpleITK object will reference to the same underlying ITK
+   * CompositeTransform. A deep-copy will be made when the object is modified.
+   */
+  CompositeTransform( const CompositeTransform & );
+
+  /** \brief Create a composite from a vector of Transform
+   *
+   * The CompositeTransform is constructed from deep copies of the Transforms.
+   *
+   * An exception is thrown if the vector is empty.
+   */
+  explicit CompositeTransform(const std::vector<itk::simple::Transform> &);
+
+  ~CompositeTransform() override;
+
+  CompositeTransform &operator=( const CompositeTransform &);
+
+  /** Name of this class */
+  std::string GetName( ) const override  { return std::string("CompositeTransform");}
+
+
+  /** \brief Removes nested composite transforms
+   *
+   * If this transform contains additional composite transforms, then these
+   * nested composite transformed are removed, while preserving the order of the
+   * regular transforms and transferring ownership to the parent
+   * CompositeTransform.
+   *
+   * Nested composite transform may not be written to a file.
+   */
+  SITK_RETURN_SELF_TYPE_HEADER FlattenTransform();
+
+  /** \brief Add a transform to the back of the stack.
+   *
+   * A deep-copy of the transform is performed. The added transform will have
+   * the optimizable parameters, while the other parameters are part of the
+   * fixed parameters.
+   */
+  SITK_RETURN_SELF_TYPE_HEADER AddTransform( Transform t );
+
+  /** \brief The number of transforms in the stack */
+  unsigned int GetNumberOfTransforms() const;
+
+  /** \brief Removed all transforms from the stack. */
+  void ClearTransforms();
+
+  /** \brief Remove the active transform at the back.
+   *
+   * If the stack is empty and exception will be thrown.
+   */
+  void RemoveTransform();
+
```

Comment:
Instead of void, possibly have the RemoveTransform return Transform, equivalent to the stack's pop?

Reply:
This follows the ITK interface and the stl stack implementation. The object is not returns as it avoid a copy if the object is not needed.